src/Servers/Kestrel/Core/src/CoreStrings.resx

```diff
@@ -716,4 +716,7 @@ For more information on configuring HTTPS see https://go.microsoft.com/fwlink/?l
   <data name="Http3ErrorControlStreamClosed" xml:space="preserve">
     <value>A control stream used by the connection was closed or reset.</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="ConnectResponseCanNotHaveBody" xml:space="preserve">
+    <value>'CONNECT' requests with a 2XX success status code cannot have a response body.</value>
```

Comment:
Nit. Let's just include the non-error status code in the exception message. Maybe someone will try setting a 101 status code or something and gets confused by this 😆.

```suggestion
    <value>Responses to 'CONNECT' requests cannot produce a response body with the successful status code: '{StatusCode}'. See IHttpExtendedConnectFeature.</value>
```

Reply:
Why reference the IHttpExtendedConnectFeature? So they can formally accept the CONNECT? In that case it should say something like "Use IHttpExtendedConnectFeature to accept the CONNECT request."

Reply:
I'm fine with "Use IHttpExtendedConnectFeature to accept the CONNECT request." 

I think the exception will be confusing for some people if the message just says you cannot produce a response body with a successful status code without indicating how to handle the request successfully.